### PR TITLE
HBN-892 Adding Purgeable Synth support

### PR DESCRIPTION
**Pull request by justinjmoses**
- Adds `PurgeableSynth` and tests, adds it to deployment where a synth has `subclass: "PurgeableSynth"`
- Add a `replace-synth` script to replace a number of synths with a new subclass
- Adds a `purge-synth` script which on `mainnet` only (due to the requirement of looking up holder addresses on Ethplorer) will lookup holders and run `purge` with them.
- Adds support to deploy to `local` network (where `ganache` is already running)
- Removes already removed synths from `deployment.json` of `kovan` and `rinkeby` and fixes `remove-synths` script
- Adds new integration test that tests build, deploy and other associated scripts. Can be extended into a testnet smoketest in the near future.
- Added CircleCI support

### CODE REVIEW

File: publish/src/commands/deploy.js [hunk 1]
```diff
@@ -358,24 +424,27 @@ module.exports = program =>
 				if (feePoolDelegateApprovals && feePool) {
 					const delegateApprovalsAddress = feePoolDelegateApprovals.options.address;
 					const feePoolOwner = await feePool.methods.owner().call();
+					const currentDelegateApprovals = await feePool.methods.delegates().call();
+
+					if (currentDelegateApprovals !== delegateApprovalsAddress) {
```
Comment (jacko125) [inline publish/src/commands/deploy.js]: @justinjmoses  We added these checks in the deploy script to the `Develop` branch. Would it be possible to base the merge off Develop branch?

File: .travis.yml [hunk 2]
```diff
@@ -3,8 +3,7 @@ language: node_js
 node_js:
   - 8.15.0
 before_install:
-  - npm i
+  - npm ci
```
[the remark this answers is not in the document]
Comment (justinjmoses) [reply .travis.yml]: well i'm going to add Circle CI support, to align this with the rest of our repos.

File: contracts/PurgeableSynth.sol [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            }
+
+        }
+
+    }
+
+    /* ========== GETTERS ========== */
+
+    function getMaxSupplyToPurge() public view returns (uint) {
+        return maxSupplyToPurgeInUSD.multiplyDecimalRound(exchangeRates.rateForCurrency("sUSD"))
```
[the remark this answers is not in the document]
Comment (justinjmoses) [reply contracts/PurgeableSynth.sol]: ah perfect - will do

File: publish/src/commands/deploy.js (hunk 1 above)
Comment (justinjmoses) [reply publish/src/commands/deploy.js]: Sorry I can't - I've overhauled all the deployment scripts. You'll have to work around me I'm afraid

File: contracts/PurgeableSynth.sol [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+-----------------------------------------------------------------
+MODULE DESCRIPTION
+-----------------------------------------------------------------
+
+Purgeable synths are a subclass of Synth that allows the owner
+to exchange all holders of the Synth back into sUSD.
+
+In order to reduce gas load on the system, and to repurpose older synths
+no longer used, purge allows the owner to
```
Comment (hav-noms) [inline contracts/PurgeableSynth.sol]: @justinjmoses finish these descriptions

File: contracts/PurgeableSynth.sol [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+These are used only for frozen or deprecated synths, and the total supply is
+hard-coded to
```
Comment (hav-noms) [inline contracts/PurgeableSynth.sol]: no opposition to 10K here

**Pull request merged**